src/main/java/team/reborn/energy/Energy.java

```diff
@@ -24,34 +30,53 @@
 	 * @return an EnergyHandler instance
 	 */
 	public static EnergyHandler of(Object object) {
-		EnergyStorage energyStorage = holderRegistry.entrySet().stream()
-			.filter(entry -> entry.getKey().test(object))
+		EnergyStorage energyStorage = PROVIDERS.stream()
+			.filter(provider -> provider.getProvidingClass().isAssignableFrom(object.getClass()) && provider.test(object))
```

Comment:
Whats the point around this filter? you havent said anything about this?

Reply:
true, I added this after, thought it would be better since this can remove all those annoying casts, can change it back if you want